pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -2013,3 +2015,43 @@ func LoadReleaseImageVerifier(config *rest.Config) (verify.Interface, error) {
 	}
 	return verify.NewFromManifests([]manifest.Manifest{m}, sigstore.NewCachedHTTPClientConstructor(sigstore.DefaultClient, nil).HTTPClient)
 }
+
+// IsErrorUpdateEvent returns true when the update event is from
+// error state in clusterdeployment.
+func IsErrorUpdateEvent(evt event.UpdateEvent) bool {
+	new, ok := evt.ObjectNew.(*hivev1.ClusterDeployment)
+	if !ok {
+		return false
+	}
+	if len(new.Status.Conditions) == 0 {
+		return false
+	}
+
+	old, ok := evt.ObjectOld.(*hivev1.ClusterDeployment)
+	if !ok {
+		return false
+	}
+
+	for _, cond := range new.Status.Conditions {
+		if controllerutils.IsConditionInDesiredState(cond) {
+			continue
+		}
```

Comment:
Wouldn't we want to update once if it was in undesired state before and now it is in desired state? We would want to give the signal that the error condition is cleared...

Reply:
the function decides if the the update event was due to error. it doesn't create signals for changes we must take care of.

in this scenario when we go from undesired to desired, the function returns false (unless there was another error change in another condition) and the update event is immediately put in the queue for reconcile.